src/core/datatypes/forte_any_chars.cpp

```diff
@@ -28,7 +28,7 @@ void CIEC_ANY_CHARS::dollarEscapeChar(std::string &paTargetBuf, char paSymbol, c
     case '\"': paTargetBuf += (paTypeID == CIEC_ANY::e_WSTRING) ? "$\"" : "\""s; break;
     default:
       if (!isprint(static_cast<unsigned char>(paSymbol)) && paTypeID == CIEC_ANY::e_STRING) {
-        std::format_to(std::back_inserter(paTargetBuf), "${:02X}", paSymbol);
+        std::format_to(std::back_inserter(paTargetBuf), "${:02X}", (unsigned char)paSymbol);
```

Comment:
Please use C++-style casts:
```suggestion
        std::format_to(std::back_inserter(paTargetBuf), "${:02X}", static_cast<unsigned char>(paSymbol));
```

Reply:
Why? This is a primitive data type, there is no ambiguity here. using a static_cast only adds noise with no advantage in return.

Reply:
Please have a look at https://stroustrup.com/bs_faq2.html#static-cast for reasons why static_cast should *always* be preferred over C-style casts.

Reply:
I don't really see any argument in that link that applies to built-in primitve data types, but that is a discussion that belongs elsewhere, so I have changed it.